ddocs/medic/views/docs_by_replication_key/map.js

```diff
@@ -48,6 +48,7 @@ function (doc) {
   };
   var value = { type: doc.type };
   switch (doc.type) {
+    case 'data_training':
```

Comment:
Making any change to a view triggers a full rebuild of the view index which is expensive. Can we keep the `doc.type` as `data_record` or will that cause issues in other places?

Reply:
yes, I'm aware of the index rebuild :( 
I tried before but it'll mix with reports and sms reports, trainings are different type of thing.
If doing it, I would have to add conditionals across the code, and make sure the implementing project's config and integrations are okay with that and not mixing in the wrong places by mistake (lists, dashboards, exports, etc)
The `data_record` is too generic, being more specific adds clarity, also the code seems to use `data_record` as an indicator of "reports".
For this FR, making a new `doc.type` seemed less invasive and conceptually straight forward because we can address "trainings" in a contained/independent way. Is that okay? Any other idea?

Reply:
I guess it depends how much these will be treated like reports, and how much they will be treated like a completely separate thing. It could be the same as reports for things like...
- marking a training Task as complete
- showing a supervisor Target of how many CHWs have completed the training
- showing trainings on the contact summary so the Supervisor can see which ones the CHW has done
- etc

If we ever do things like that it'll be much easier if it's just another report.

However you're right, if we want to _exclude_ the completed trainings from these features then we'll need to add a bunch of statements to filter them out.

My personal view is we should treat them as regular reports so app developers can build those features if they want to, while including exceptions to filter them out only where necessary.

Can you think of places we definitely don't want to return training reports?

Reply:
I met with the team today, we decided to just let training form behave as the system allows it (just like reports), we won't be filtering trainings from anywhere and they will be `data_record`. We'll check user feedback and reevaluate if we need to modify something in the training behaviour